src/main/java/io/nats/client/impl/NatsConnectionWriter.java

```diff
@@ -58,7 +61,12 @@ class NatsConnectionWriter implements Runnable {
         writerLock = new ReentrantLock();
 
         this.running = new AtomicBoolean(false);
-        this.reconnectMode = new AtomicBoolean(sourceWriter != null);
+        if (sourceWriter == null) {
+            mode = new AtomicReference<>(Mode.Normal);

```

Comment:
source writer is null on a new connection - it's added later.
source writer is not null on a reconnect, because the old one is passed to the new connection so the new connection can get the old connection's normal queue